test/test_cpp_extensions.py

```diff
@@ -622,6 +622,17 @@ def test_set_default_type_also_changes_aten_default_type(self):
         finally:
             torch.set_default_dtype(initial_default)
 
+    def test_compilation_error_formatting(self):
+        # Test that the missing-semicolon error message has linebreaks in it. 
+        # This'll fail if the message has been munged into a single line.
+        # It's hard to write anything more specific as every compiler has it's own
+        # error formatting.
+        pattern = r'.*[\n\r].*'
```

Comment:
Hmm are you sure that this looks exactly for `\n` or `\r` (written verbosely) in the message and not for any of `\`, `n` or `r`? I can't get this pattern to work properly with `re.match` 😕

Reply:
This is the behaviour I expect and the behaviour I see:
```python
pattern = r'.*[\n\r].*'
re.match(pattern, 'one n two') # none
re.match(pattern, 'one \ two') # none
re.match(pattern, 'one \n two') # match
````
I'm on Python 3; is something different on 2?

Reply:
This looks right to me.

Reply:
Shouldn't be be guarding against `\n` appearing verbosely (i.e. a `"\\n"` Python string) instead? Right now you're checking that the output contains a newline which is likely to be trivially satisfied because of some other prints?

Reply:
OK, I see our misunderstanding. I think you expect a more stringent test than I've implemented. Testing that the output contains a newline is actually the intent; it means the compiler string isn't being munged into a single line because it was cast directly from a bytestring rather than decoded.

```python
pattern = r'.*[\n\r].*'
message = b'\n'
re.match(pattern, str(message)) # bad cast, returns none
re.match(pattern, message.decode()) # good decode, returns match
```

Reply:
Well I think my test is still better. There can be newlines that come from formatting of other parts of the error message, but we should never see "\\n" in an error message because that means that we've clearly messed up. This also checks out with your assumption that the compiler error message contains newlines, because then that sequence of characters is guaranteed to appear if we haven't formatted it correctly.

Reply:
Alter this to show me what you're talking about?
```python
pattern = r'.*[\n\r].*'
message = b'\n'
re.match(pattern, str(message)) # bad cast, returns none
re.match(pattern, message.decode()) # good decode, returns match
```
I just tried `r'.*[\\n\\r].*'` and that doesn't work, while `'.*[\\n\\r].*'` does work but is equivalent to the current regex. Raw strings already escape the backslash.

Reply:
Ok, basically what I meant is that we should also check that the message _does not match the following regex pattern_: `r'.*(\\n|\\r).*'` (i.e. if you try to print the error message string both `\n` and `\r` _will not appear on the user's screen_).

Reply:
Pushed. Thanks!